samples/Ark.ReferenceProject/Core/Ark.Reference.Core.Tests/Features/BookPrintProcessSteps.cs

```diff
@@ -0,0 +1,214 @@
+                return _client.LastStatusCodeIsSuccess()
+                    ? _client.ReadAs<BookPrintProcess.V1.Output?>()
+                    : null;
+            });
+        }
+
+        [When(@"I wait for the print process to fail")]
+        public async Task WhenIWaitForThePrintProcessToFail()
+        {
+            // Wait for bus/outbox to be idle
+            await _testHost.ThenIWaitBackgroundBusToIdleAndOutboxToBeEmpty().ConfigureAwait(false);
+
+            // Poll the GET endpoint until error status using Polly
+            var policy = Policy
+                .HandleResult<BookPrintProcess.V1.Output?>(r => r?.Status != BookPrintProcessStatus.Error)
+                .WaitAndRetry(30, _ => TimeSpan.FromSeconds(1));
+
+            _currentPrintProcess = policy.Execute(() =>
+            {
+                _client.Get($"{_controllerName}/{_currentPrintProcessId}");
+                return _client.LastStatusCodeIsSuccess()
+                    ? _client.ReadAs<BookPrintProcess.V1.Output?>()
+                    : null;
+            });
+        }
+
+        [When(@"I wait for the IFailed handler to process the error")]
+        public async Task WhenIWaitForTheIFailedHandlerToProcessTheError()
+        {
+            // Same pattern as waiting for failure
+            await WhenIWaitForThePrintProcessToFail().ConfigureAwait(false);
+        }
+
+        [When(@"I retrieve the print process status")]
+        public void WhenIRetrieveThePrintProcessStatus()
+        {
+            _client.Get($"{_controllerName}/{_currentPrintProcessId}");
+            _client.ThenTheRequestSucceded();
+            _currentPrintProcess = _client.ReadAs<BookPrintProcess.V1.Output>();
+        }
+
+        [Then(@"the print process should be created with status ""(.*)""")]
+        public void ThenThePrintProcessShouldBeCreatedWithStatus(string status)
+        {
+            _currentPrintProcess.Should().NotBeNull();
+            _currentPrintProcess!.Status.ToString().Should().Be(status);
+        }
+
+        [Then(@"the print process progress should be (.*)")]
+        public void ThenThePrintProcessProgressShouldBe(double expectedProgress)
+        {
+            _currentPrintProcess.Should().NotBeNull();
+            _currentPrintProcess!.Progress.Should().BeApproximately(expectedProgress, 0.01);
+        }
+
+        [Then(@"I should get a (.*) Bad Request response")]
+        public void ThenIShouldGetABadRequestResponse(int statusCode)
+        {
+            _client.ThenTheRequestFailsWith((System.Net.HttpStatusCode)statusCode);
+        }
+
+        [Then(@"the error should indicate ""(.*)""")]
+        public void ThenTheErrorShouldIndicate(string ruleCode)
+        {
+            _lastErrorContent.Should().NotBeNullOrEmpty();
+            _lastErrorContent.Should().Contain(ruleCode);
+        }
+
+        [Then(@"the print process status should be ""(.*)""")]
+        public void ThenThePrintProcessStatusShouldBe(string status)
```

Comment:
replace this with the common entity comparison Verb with Table and CompareToInstance as described in the other comment